bootstrap/src/main/java/io/airlift/bootstrap/FuzzyMatcher.java

```diff
@@ -0,0 +1,51 @@
+package io.airlift.bootstrap;
+
+import com.google.common.collect.ImmutableList;
+import me.xdrop.fuzzywuzzy.FuzzySearch;
+
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+import static com.google.common.base.Verify.verify;
+import static java.lang.Math.min;
+import static java.util.Comparator.comparingInt;
+import static java.util.Objects.requireNonNull;
+
+class FuzzyMatcher
+{
+    private FuzzyMatcher() {}
+
+    public static List<String> findSimilar(String key, Set<String> candidates, int count)
+    {
+        if (candidates.contains(key)) {
+            throw new IllegalArgumentException("Key '%s' is already present in the %s candidates".formatted(key, candidates));
+        }
+
+        PriorityQueue<Match> matchQueue = new PriorityQueue<>(3, comparingInt(Match::ratio).reversed());
+        for (String seenProperty : candidates) {
+            int ratio = FuzzySearch.ratio(seenProperty, key);
+            if (ratio > 75) { // At least 75% similarity
+                matchQueue.add(new Match(seenProperty, ratio));
+            }
+        }
+
+        int elements = min(count, matchQueue.size());
+        ImmutableList.Builder<String> builder = ImmutableList.builderWithExpectedSize(elements);
+
+        for (int i = 0; i < elements; i++) {
+            builder.add(matchQueue.poll().key());
+        }
```

Comment:
Any reason not to go with a simpler implementation that selects all the elements that are similar (> 75%), sorts them, and then picks the top N? E.g.,

```java
        return candidates.stream()
                .map(candidate -> new Match(candidate, FuzzySearch.ratio(candidate, key)))
                .filter(match -> match.ratio() > 75)
                .sorted(comparingInt(Match::ratio).reversed())
                .limit(count)
                .map(Match::key)
                .collect(ImmutableList.toImmutableList());
```

Reply:
Makes sense, thanks!